tests/testsuite/build.rs

```diff
@@ -1563,6 +1579,232 @@ fn crate_env_vars() {
     }
 }
 
+#[cargo_test]
+fn nightly_cargo_workspace_dir_env_var_with_workspace() {
+    Package::new("bar", "0.1.0")
+        .file("src/lib.rs", "#[test] fn bar() {}")
+        .file(
+            "tests/env.rs",
+            r#"
+            use std::path::Path;
+
+            #[test]
+            fn env() {
+                assert!(Path::new(option_env!("CARGO_WORKSPACE_DIR").unwrap()).join(file!()).exists());
+                // assert_eq!(std::fs::canonicalize(option_env!("CARGO_WORKSPACE_DIR").unwrap()).unwrap().display().to_string(), option_env!("CARGO_MANIFEST_DIR").unwrap());
```

Comment:
@epage you suggested that for published crates, we should get the same directory as `CARGO_MANIFEST_DIR`. This is not the case here.
```
thread 'env' panicked at 'assertion failed: `(left == right)`
  left: `"/Users/yerke/dev/rust/cargo/target/tmp/cit/t0/foo"`,
 right: `"/Users/yerke/dev/rust/cargo/target/tmp/cit/t0/home/.cargo/registry/src/-c29b6f5e91792042/bar-0.1.0"`', /Users/yerke/dev/rust/cargo/target/tmp/cit/t0/home/.cargo/registry/src/-c29b6f5e91792042/bar-0.1.0/tests/env.rs:7:17
```

`CARGO_MANIFEST_DIR` from inside the published crate's test still seems to point to the published crate's manifest directory, whereas `CARGO_WORKSPACE_DIR` does point to the our current package's workspace directory.

Reply:
I think I might be misunderstanding something.

`CARGO_WORKSPACE_DIR` should be pointing to the `CARGO_MANIFEST_DIR` for what is being built.  We should not be exposing details of the dependent.  I was concerned that this was not being followed and was trying to point out that this was a potential failure case of the current implementation,.

Reply:
Got it. Yes, I did misunderstand what you meant by it.

Reply:
I updated the PR and it should now behave how you envisioned it (I think).